localstack/services/awslambda/invocation/docker_runtime_executor.py

```diff
@@ -264,25 +282,25 @@ def start(self, env_vars: dict[str, str]) -> None:
             or self.function_version.config.package_type != PackageType.Zip
         ):
             CONTAINER_CLIENT.copy_into_container(
-                self.id, f"{str(get_runtime_client_path())}/.", "/"
+                self.container_name, f"{str(get_runtime_client_path())}/.", "/"
             )
         if not config.LAMBDA_PREBUILD_IMAGES:
             # copy_folders should be empty here if package type is not zip
             for source, target in container_config.copy_folders:
-                CONTAINER_CLIENT.copy_into_container(self.id, source, target)
+                CONTAINER_CLIENT.copy_into_container(self.container_name, source, target)
 
-        CONTAINER_CLIENT.start_container(self.id)
+        CONTAINER_CLIENT.start_container(self.container_name)
```

Comment:
Sidenote: If we want to protect ourselves from external renaming in the future, we could consider using the [Container.id](https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.id) / [CLI container id](https://docs.docker.com/engine/reference/commandline/run/#cidfile) instead of the name but the name seems to work fine for now.